<launch>
    <arg name="dof" default="1" />
    <arg name="target_object" default="syvess" />
    <arg name="present_jointstate_topic" default="/motors/present_states" />
    <arg name="goal_jointstate_topic" default="/motors/goal_states" />
    <arg name="reset_srv_name" default="/motors/reset" />

    <arg name="camera" default="camera" />
    <arg name="device" default="/dev/video0" />

    <!-- image acquisition from camera -->
    <include file="$(find trackerrr)/launch/camera.launch" >
        <arg name="camera" value="$(arg camera)" />
        <arg name="device" value="$(arg device)" />
    </include>

    <!-- object detection from camera images -->
    <include file="$(find darknet_ros)/launch/darknet_ros.launch" />

    <!-- object tracker -->
    <node pkg="trackerrr" type="yolotracker_node" name="yolotracker" output="screen">
        <param name="dof" value="$(arg dof)" />
        <param name="goal_jointstate_topic" value="$(arg goal_jointstate_topic)" />
        <param name="reset_srv_name" value="$(arg reset_srv_name)" />
        <param name="target_object" value="$(arg target_object)" />
        <param name="image_topic" value="/$(arg camera)/image_rect" />
        <param name="camera" value="$(arg camera)" />
    </node>

    <!-- dynamixel motor controller -->
    <include file="$(find trackerrr)/launch/dynamixel_controller.launch">
        <arg name="dof" value="$(arg dof)" />
        <arg name="present_jointstate_topic" value="$(arg present_jointstate_topic)" />
        <arg name="goal_jointstate_topic" value="$(arg goal_jointstate_topic)" />
    </include>

    <!-- viz of current robot joint states -->
    <include file="$(find trackerrr)/launch/viz_robot.launch">
        <arg name="dof" value="$(arg dof)" />
        <arg name="present_jointstate_topic" value="$(arg present_jointstate_topic)" />
    </include>
    
    <!-- view image output -->
    <node pkg="rqt_image_view" type="rqt_image_view" name="rqt_image_view" args="/darknet_ros/detection_image" />
</launch>